flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/GenericWriteAheadSink.java

```diff
@@ -77,63 +86,96 @@ public GenericWriteAheadSink(CheckpointCommitter committer, TypeSerializer<IN> s
 	public void open() throws Exception {
 		super.open();
 		committer.setOperatorId(id);
-		committer.setOperatorSubtaskId(getRuntimeContext().getIndexOfThisSubtask());
 		committer.open();
-		cleanState();
-		checkpointStreamFactory =
-				getContainingTask().createCheckpointStreamFactory(this);
+
+		checkpointStreamFactory = getContainingTask()
+			.createCheckpointStreamFactory(this);
+
+		cleanRestoredHandles();
 	}
 
 	public void close() throws Exception {
 		committer.close();
 	}
 
 	/**
-	 * Saves a handle in the state.
+	 * Called when a checkpoint barrier arrives.
+	 * Closes any open streams to the backend and marks them as pending for
+	 * committing to the final output system, e.g. Cassandra.
```

Comment:
to the final output system -> to the external backend; remove mentioning of cassandra.